iron-location.html

```diff
@@ -171,36 +171,40 @@
         // one when we set this.hash. Likewise for query.
         this._dontUpdateUrl = true;
         this._hashChanged();
-        this.path = window.location.pathname;
-        this.query = window.location.search.substring(1);
+        this.path = window.decodeURIComponent(window.location.pathname);
+        this.query = window.decodeURIComponent(
+            window.location.search.substring(1));
         this._dontUpdateUrl = false;
         this._updateUrl();
       },
       _getUrl: function() {
-        var url = this.path;
-        var query = this.query;
-        if (query) {
-          url += '?' + query;
+        var partiallyEncodedPath = window.encodeURI(
+            this.path).replace(/\#/g, '%23').replace(/\?/g, '%3F');
+        var partiallyEncodedQuery = '';
+        if (this.query) {
+          partiallyEncodedQuery = '?' + window.encodeURI(
+              this.query).replace(/\#/g, '%23');
```

Comment:
Fragment identifier is the hash yeah? My thinking here is that ideally the user of iron-location wouldn't have to think about encoding, and that when they set a property on an iron-location that they intend only to set the corresponding part of the URL, even if it includes a character that would otherwise signal the end of that part. So when someone does:

```javascript
ironLocation.query = '?tweet=@foo bar #baz';
```

They should get `http://example.com/?tweet=@foo%20bar%20%23baz` as their url. That has the nice property that after refreshing the page, `ironLocation.query === '?tweet=@foo bar #baz'` is true.